lib/txgh/handlers/transifex/hook_handler.rb

```diff
@@ -20,21 +20,25 @@ def initialize(options = {})
         def execute
           logger.info(resource_slug)
 
-          if tx_config
-            if tx_resource
-              committer = ResourceCommitter.new(project, repo, logger)
-              committer.commit_resource(tx_resource, branch, language)
+          if supported_language?
+            if tx_config
+              if tx_resource
+                committer = ResourceCommitter.new(project, repo, logger)
+                committer.commit_resource(tx_resource, branch, language)
 
-              respond_with(200, true)
+                respond_with(200, true)
+              else
+                respond_with_error(
+                  404, "Could not find resource '#{resource_slug}' in config"
+                )
+              end
             else
               respond_with_error(
-                404, "Could not find resource '#{resource_slug}' in config"
+                404, "Could not find configuration for branch '#{branch}'"
               )
             end
           else
-            respond_with_error(
-              404, "Could not find configuration for branch '#{branch}'"
-            )
+            respond_with(304, true)
           end
         end
```

Comment:
I like that, but it makes the individual lines too long. Will try with non-inline conditionals.